sklearn/datasets/_lfw.py

```diff
@@ -72,24 +72,38 @@
 #
 
 
-def _check_fetch_lfw(data_home=None, funneled=True, download_if_missing=True):
+def _check_fetch_lfw(data_home=None, funneled=True, download_if_missing=True, show_progress=False):
     """Helper function to download any missing LFW data"""
-
+    
     data_home = get_data_home(data_home=data_home)
     lfw_home = join(data_home, "lfw_home")
 
     if not exists(lfw_home):
         makedirs(lfw_home)
 
+    progress = None
+    task = 0
+
+    if show_progress:
+        try:
+            from rich.progress import Progress
+            progress = Progress()
+            task = progress.add_task("[red]Downloading metadatas...", total=len(TARGETS))
+            progress.start()
+        except ImportError:
+            pass
+    
     for target in TARGETS:
         target_filepath = join(lfw_home, target.filename)
         if not exists(target_filepath):
             if download_if_missing:
                 logger.info("Downloading LFW metadata: %s", target.url)
-                _fetch_remote(target, dirname=lfw_home)
+                _fetch_remote(target, dirname=lfw_home, progress=progress)
```

Comment:
Do you think that we could limit changes by only having the code related to downloading in `_fetch_remote` such that we don't need to modify the code of any fetcher but rather that any fetcher using `_fetch_remote` would benefit from the change?

Reply:
If only one file is fetched, the only modification of the `_fetch_remote` function is meaningful, but I don't think it's meaningful to change only the `_fetch_remote` code when fetching multiple files. We don't know how many files we downloaded, and  how much the actual progress is.

Reply:
> but I don't think it's meaningful to change only the _fetch_remote code when fetching multiple files

I think that in terms of UX, we want to show progress on long downloading task to inform the user that something happen. Having an additional progress bar for a task that take a couple of millisecond to download of metatadata file of a couple kilobytes will not helpful.

Since `_fetch_remote` is used to download all datasets file, this is the reason that I would start to only focus on this file.

Reply:
If so, should I add a progress bar to all fetchers that use `_fetch_remote`(like `_check_fetch_lfw`, `fetch_california_housing`, ..)  instead of modifying the `_fetch_remote` function?

Reply:
No you should only modify `_fetch_remote`